### ISSUE TITLE
Not have the document rely on the algorithm defined in Accessible Name and Description: Computation and API Mappings 1.1

### ISSUE THREAD

**Issue by ollie-iterators** [w3c/html-aam#393]
I think that the document could be improved by making it so that the user does not need to use the algorithm in [Accessible Name and Description: Computation and API Mappings 1.1](https://www.w3.org/TR/accname-aam-1.1/).

That would mean, that there are some attributes in the algorithm that are not supposed to be used in certain elements. For example, the "alt" attribute is not supposed to be included in SVG elements, so that could not be included in the steps for accessible name/description generation for SVG elements.

**Comment by scottaohara** [comment]
The naming algorithm section definitely needs some cleanup/updates.  but i'm not sure i follow your ask?  being that right now the section in this document indicates the mechanisms to name particular elements (svg naming being defined in its own aam spec), an image and image button note they can use the alt, where as other elements do not list the alt attribute as it is not applicable to them.

**Comment by ollie-iterators** [comment]
I meant splitting up the algorithm based on the type of element that was being tested.

**Comment by scottaohara** [comment]
that is already how this spec is written?  we refer to the accName spec as it is the primary specification for determining this, but this spec already defines the variations for the host language naming features (save those features that we already have issues filed for) on a per element (or element grouping) basis.  

So, is there something specific you are looking to have changed about the way the spec already handles this?  I would be very happy to hear suggestions on improvement ideas for clarity.

**Comment by ollie-iterators** [comment]
Ok, as a hypothetical example, I am going to use Images to describe how I think the algorithm should be organized.

For this example, I will assume that images are either: SVG Images or Non SVG Images. If the Images are decorative, then they do not need any accessible name so that could be included also.

SVG Images use the title or desc html tags to have accessible names. This is different than how Non SVG Images get Accessible Names, because they can use the following attributes:

- alt
- aria-label
- aria-labelledby
- aria-describedby

Non SVG Images can also use the following elements:

- figure
- figcaption

The information for SVG Images was found from this URL: https://css-tricks.com/accessible-svgs/
The information for Non SVG Images was found from this URL: https://usability.yale.edu/web-accessibility/articles/images#image-guidelines

The algorithm for getting accessible names for SVG Images would be different than the algorithm for getting accessible names for Non SVG Images because they use different attributes.

So in conclusion, I would want an algorithm for Images (separated by type of image) that includes all the attributes and elements that are necessary for the type of images that are having the accessible name looked for, and so on for the other types of elements on the page (Links, Buttons, Tables, ...)

**Comment by stevefaulkner** [comment]
@ollie-iterators . HTML AAM's  intended audience is browser implementers. The goal of the specification is to define a set of implementation requirements to which browsers adhere to in order to achieve interoperability.  As far as SVG goes [the SVG element](https://svgwg.org/svg2-draft/struct.html#SVGElement) is not defined in HTML it's defined in the SVG spec and any special processing algorithms are therefore defined in the [SVG AAM](https://w3c.github.io/svg-aam/#mapping_additional)

**Comment by scottaohara** [comment]
But again, @ollie-iterators, this spec already outlines the naming algorithms per element and as I mentioned and steve repeated, SVG AAM defines the naming mechanisms for SVGs.

Have you reviewed the naming and description section of this spec yet?

As an aside, figure/figcaption _do not contribute_ to the naming of images.  They have their own section for naming (and that's in process of being re-implemented by browsers per an open PR on this subject)(

**Comment by ollie-iterators** [comment]
Ok, so it seems that it is already okay.
